File: Backend/renderer/src/test/java/io/sparkled/util/matchers/SparkledMatchers.java
package io.sparkled.util.matchers;

import io.sparkled.model.render.RenderedChannel;
import io.sparkled.model.render.RenderedFrame;
import org.hamcrest.Matcher;

public class SparkledMatchers {
    public static Matcher<RenderedChannel> hasLeds(int[][] leds) {
        return new RenderedChannelLedMatcher(leds);
    }

    public static Matcher<RenderedFrame> hasLeds(int[] leds) {
        return new RenderedFrameLedMatcher(leds);
    }
}

File: renderer/src/test/kotlin/io/sparkled/util/matchers/SparkledMatchers.kt
package io.sparkled.util.matchers

import io.sparkled.model.render.RenderedStagePropData
import org.hamcrest.Matcher

object SparkledMatchers {
    fun hasRenderedFrames(vararg leds: IntArray): Matcher<RenderedStagePropData> {
        return RenderedStagePropDataLedMatcher(arrayOf(*leds))
    }
}
